interfaces/apparmor/backend.go

```diff
@@ -272,7 +272,9 @@ func (b *Backend) Setup(snapInfo *snap.Info, opts interfaces.ConfinementOptions,
 	if err != nil {
 		return fmt.Errorf("cannot obtain apparmor specification for snap %q: %s", snapName, err)
 	}
+	spec.(*Specification).snapName = snapInfo.Name()
 	spec.(*Specification).AddSnapLayout(snapInfo)
+	spec.(*Specification).snapName = ""
```

Comment:
I think you need a comment here cause this looks wrong. It seems that you are setting snapName so that ExpandPathVariables does what you want, then set it back to avoid side effects. It looks like one of these is redundant and I fear someone will ask the question later and remove one. If this assessment is wrong, please remove one. :)

Reply:
I'm doing this so that AddUpdateNS() is not a no-op. In retrospective I think we can drop the snapName variable from apparmor.Specification and just record the snap-update-ns snippets in a simple list.